linalg/dual.hpp

```diff
@@ -342,48 +349,60 @@ dual<value_type, gradient_type> tan(dual<value_type, gradient_type> a)
 template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> atan(dual<value_type, gradient_type> a)
 {
+   using std::atan;
    return {atan(a.value), a.gradient / (value_type{1} + a.value * a.value)};
 }
 
 /** @brief implementation of exponential function for dual numbers */
 template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> exp(dual<value_type, gradient_type> a)
 {
-   return {std::exp(a.value), std::exp(a.value) * a.gradient};
+   using std::exp;
+   return {exp(a.value), exp(a.value) * a.gradient};
 }
 
 /** @brief implementation of the natural logarithm function for dual numbers */
 template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> log(dual<value_type, gradient_type> a)
 {
-   return {std::log(a.value), a.gradient / a.value};
+   using std::log;
+   return {log(a.value), a.gradient / a.value};
 }
 
 /** @brief implementation of `a` (dual) raised to the `b` (dual) power */
 template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> pow(dual<value_type, gradient_type> a,
                                     dual<value_type, gradient_type> b)
 {
+   using std::log;
+   using std::pow;
    value_type value = pow(a.value, b.value);
```

Comment:
I'm talking about lines like this. Here, you only need `pow(double,double)`, right?

Reply:
Or, do you actually need nested dual types?

Reply:
I do need it for nested dual types - the nesting is used for second derivatives and is a functionality which I need.

Reply:
Oh, okay, why didn't you say so earlier? 😄

This means you want this `pow` call to resolve to `std::pow` when using it with `value_type=double`, and to `mfem::internal::pow` when using it with `value_type=dual<double,double>`.

By the way, this fix seems to be necessary with Clang as well, not just with GCC, even without the nested dual types.